### ISSUE TITLE
Issues when using MySQL 8 with ANSI_QUOTES "Global SQL mode"

### ISSUE THREAD

**Issue by trajche** [deliciousbrains/wp-amazon-s3-and-cloudfront#532]
ANSI_QUOTES mode in MySQL 8 (which is enabled by default on DigtialOcean MySQL 8 Managed databases) treats " as an identifier quote character (like the ` quote character) and not as a string quote character.

To fix this, I've changed the following in `classes/items/media-library-item.php`:
```
if ( $exact_match ) {
   $sql .= ' AND (items.source_path IN ("' . join( '","', $paths ) . '")';
   $sql .= ' OR items.original_source_path IN ("' . join( '","', $paths ) . '"))';
} else {
```

to:
```
if ( $exact_match ) {
   $sql .= ' AND (items.source_path IN (\'' . join( '\',\'', $paths ) . '\')';
   $sql .= ' OR items.original_source_path IN (\'' . join( '","', $paths ) . '\'))';
} else {
```
in the get_by_source_path function line 252.

**Comment by ianmjones** [comment]
Not sure whether WordPress supports that at the moment, but we've seen this before in support and are looking to fix things up for a future release.

In the meantime, I think you can use the [incompatible_sql_modes](https://developer.wordpress.org/reference/hooks/incompatible_sql_modes/) filter to disable that mode a runtime.

**Comment by ianmjones** [comment]
Fixed in WP Offload Media (Lite) 2.4.
